### ISSUE TITLE
"Bug Detected" Error Popup

### ISSUE THREAD

**Issue by sunnybubblegum** [mypaint/mypaint#926]
#### Description of the problem

Hi, I'm on a fresh install of the new Ubuntu 18.04 LTS. I just downloaded MyPaint not long ago through the Ubuntu Software application. It's been working great, except suddenly out of nowhere, this error popup appeared.

![mypaint-error-popup](https://user-images.githubusercontent.com/1007764/39354460-3d9343a6-49d0-11e8-846f-13b758eab792.jpg)

#### Basic system details

MyPaint version: 1.2.0+gitexport.f62444e
Operating system: Ubuntu 18.04
Desktop environment: GNOME (the default, non-Wayland version)

#### Steps to reproduce

1. Start MyPaint.
2. Open an existing .ora file. The project I opened was from a previous, up-to-date version of MyPaint running on Ubuntu 14.04.5.
3. Work normally. The only thing I really did with the file was change the framing a little bit. Didn't save.
4. The error popup appears. The program still runs, and it appears I can dismiss it and continue working.

#### Backtraces or error messages

```
**Bug Detected**

**A programming error has been detected.**

You may be able to ignore this error and carry on working, but you should probably save your work soon.

Please tell the developers about this using the issue tracker if no-one else has reported it yet.
```

**Comment by sunnybubblegum** [comment]
It just occurred to me to include the output under the popup's "Details..." expansion area. Here it is.

```python
Exception while analyzing the exception.
Traceback (most recent call last):
  File "/usr/share/mypaint/gui/document.py", line 619, in button_release_cb
    return CanvasController.button_release_cb(self, tdw, event)
  File "/usr/share/mypaint/gui/document.py", line 114, in button_release_cb
    result = mode.button_release_cb(tdw, event)
  File "/usr/share/mypaint/gui/freehand.py", line 410, in button_release_cb
    self.doc.input_stroke_ended(event)
  File "/usr/share/mypaint/lib/observable.py", line 284, in __call__
    observer(observed, *args, **kwargs)
  File "/usr/share/mypaint/lib/observable.py", line 467, in __call__
    self._observer_func(observer, observed, *args, **kwargs)
  File "/usr/share/mypaint/gui/brushmanager.py", line 761, in _input_stroke_ended_cb
    hb_info = hb.brushinfo
  File "/usr/share/mypaint/gui/brushmanager.py", line 943, in get_brushinfo
    self._ensure_settings_loaded()
  File "/usr/share/mypaint/gui/brushmanager.py", line 1185, in _ensure_settings_loaded
    self.load()
  File "/usr/share/mypaint/gui/brushmanager.py", line 1127, in load
    self._load_settings()
  File "/usr/share/mypaint/gui/brushmanager.py", line 1149, in _load_settings
    self._brushinfo.load_from_string(brushinfo_str)
  File "/usr/share/mypaint/lib/brush.py", line 181, in load_from_string
    self.from_json(settings_str)
  File "/usr/share/mypaint/lib/brush.py", line 150, in from_json
    brush_def = json.loads(json_string)
  File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 367, in decode
    raise ValueError(errmsg("Extra data", s, end, len(s)))
ValueError: Extra data: line 247 column 2 - line 248 column 2 (char 6313 - 6326)
```

**Comment by briend** [comment]
It sounds like a corrupt brush.  Does it only happen with a particular brush selected?  Are you using the "pick stroke" feature to grab the brush from the canvas of that ora document?

**Comment by sunnybubblegum** [comment]
Hm, I couldn't be sure.

Am noticing that the popup first appears when laying down the first brush stroke after opening the program. Unsure which brush is selected on startup, or how to find out.

The popup then appears again upon trying to quit the program.

Am not using the "Pick Stroke" feature (didn't even know of it before you mentioned it).

Have also found that the error occurs on new projects too, not just existing ones from older MyPaint versions.

Can I be of further help?

**Comment by briend** [comment]
Could you try a newer version of MyPaint? http://mypaint.org/downloads/

**Comment by jplloyd** [comment]
Issue reporter not responding, and no similar reports have been issued. Hopefully this was a rare combination of circumstances (perhaps fixed in 1.2.1). Make a comment if you want this reopened. If the offending files are still available, we can investigate it.
